mountpoint-s3-crt/src/checksums/crc.rs

```diff
@@ -0,0 +1,34 @@
+use mountpoint_s3_crt_sys::*;
+
+/// Calculate CRC32 checksum of the data in the given buffer, append to the previous checksum if provided.
+pub fn checksums_crc32(buf: &[u8], previous_checksum: Option<u32>) -> u32 {
+    // SAFETY: we pass a valid buffer to the CRT, and trust
```

Comment:
If CRT is actually conduct read-only to the given buffer, we can be more affirmative here.

Something like ""CRT reads from the buffer's boundary only"